index.bs

```diff
@@ -552,12 +552,71 @@ spec: XHR; urlPrefix: https://xhr.spec.whatwg.org/
 
   Cross-origin messaging via <a>`postMessage`</a> requires that the recipient be
   able to see the suborigin namespace of the message sender so it can make an
-  appropriate access control decision. When a message is sent from a suborigin
-  namespace, the receiver has the `event.origin` value set to the serialized
-  suborigin value instead of the serializied origin, as described in
-  [[#serializing]]. Additionally, a new `suborigin` property must be added to
-  the `MessageEvent` given to the receiver which contains the suborigin
-  namespace value.
+  appropriate access control decision. We introduce a new dictionary
+  `EventInitiator` that holds information about both the origin and
+  the suborigin namespace:
+
+  <pre class="idl">
+    dictionary EventInitiator {
+      USVString origin = "";
+      USVString? suborigin = null;
+    };
+  </pre>
+
+  Issue: We're monkey-patching HTML, submit a PR when we're sure this is the
+  general direction we want to go in.
+
+  And we extend the definition of the MessageEvent and MessageEventInit as follows:
+
+  <pre class="idl">
+    interface MessageEvent {
+      readonly attribute USVString? origin;
+      readonly attribute EventInitiator initiator;
+      void initMessageEvent(DOMString type, optional boolean bubbles = false, optional boolean cancelable = false, optional any data = null, optional EventInitiator? initator = null, optional DOMString lastEventId = "", optional MessageEventSource? source = null, optional sequence&lt;MessagePort&gt; ports = []);
+    };
+
+    dictionary MessageEventInit {
+      EventInitiator? initiator = null;
+    };
+  </pre>
+
+  Similarly, we add a new overloaded version of postMessage to the Window interface:
+
+  <pre class="idl">
+    partial interface Window {
+      void postMessage(any message, optional EventInitiator targetOrigin, optional sequence&lt;object&gt; transfer = []);
+    };
+  </pre>
+
+  Those new interfaces can also be used to communicate with browsing contexts
+  that aren't in a suborigin namespace. Furthermore, the `origin` attribute of
+  an `MessageEvent` will be `null` (the value, not the string) if the
```

Comment:
correct.

You could just inject a toString method on the prototype of EventInitiator (or whatever the interface will be called) that serializes the origin + suborigin for you.